src/modeles/mesuresGenerales.js

```diff
@@ -13,19 +13,29 @@ class MesuresGenerales extends ElementsConstructibles {
   }
 
   parStatut() {
+    const accumuleMesure = (acc, mesure) => {
+      const mesureReference = this.referentiel.mesure(mesure.id);
+      acc[mesure.statut][mesureReference.categorie] ||= [];
+      acc[mesure.statut][mesureReference.categorie].push({
+        description: mesure.descriptionMesure(),
+        indispensable: mesure.estIndispensable(),
+        modalites: mesure.modalites,
+      });
+      return acc;
+    };
+
     let mesuresParStatut = { fait: {}, enCours: {}, nonFait: {} };
-    mesuresParStatut = this.toutes()
-      .filter((mesure) => mesure.statutRenseigne())
-      .reduce((acc, mesure) => {
-        const mesureReference = this.referentiel.mesure(mesure.id);
-        acc[mesure.statut][mesureReference.categorie] ||= [];
-        acc[mesure.statut][mesureReference.categorie].push({
-          description: mesure.descriptionMesure(),
-          indispensable: mesure.estIndispensable(),
-          modalites: mesure.modalites,
-        });
-        return acc;
-      }, mesuresParStatut);
+
+    const mesuresRenseignes = this.toutes().filter((mesure) => mesure.statutRenseigne());
+
+    mesuresParStatut = mesuresRenseignes
+      .filter((mesure) => mesure.estIndispensable())
+      .reduce(accumuleMesure, mesuresParStatut);
+
+    mesuresParStatut = mesuresRenseignes
+      .filter((mesure) => mesure.estRecommandee())
+      .reduce(accumuleMesure, mesuresParStatut);
```

Comment:
Pour éviter la double itération, regarde du côté de `sort()`.

Ça pourrait donner qqch comme 


```diff
const indispensablesDabord = (a, b) => { ... } 

mesuresRenseignes
-     .filter((mesure) => mesure.estIndispensable())
+     .sort(indispensableDabord)
      .reduce(accumuleMesure, mesuresParStatut);
```